codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Exception.java

```diff
@@ -47,6 +63,13 @@ public Http2Error error() {
     }
 
     /**
+     * If a connection error should attempt a graceful shutdown.
+     * @return {@code true} if a connection error should attempt a graceful shutdown. {@code false} otherwise.
+     */
+    public boolean gracefulShutdown() {
```

Comment:
nit: Make `shouldShutdownGracefully`?  Also I wonder if it would make sense to make this an enum:

```java
enum ShutdownMode {
    NO_SHUTDOWN_REQUIRED,
    SHUTDOWN_GRACEFULLY,
    SHUTDOWN_NOW
}
```

The default for stream errors would be `NO_SHUTDOWN_REQUIRED` and the default for connection errors would be `SHUTDOWN_GRACEFULLY`.  WDYT?